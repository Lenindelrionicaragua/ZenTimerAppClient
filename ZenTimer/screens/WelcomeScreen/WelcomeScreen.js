import React from "react";
import { StatusBar } from "react-native";

import {
  StyledContainer,
  InnerContainer,
  PageTitle,
  SubTitle,
  StyledFormArea,
  StyledButton,
  ButtonText,
  Line,
  WelcomeContainer,
  WelcomeImage,
  Avatar,
} from "./WelcomeScreenStyles";

const WelcomeScreen = () => {
  return (
    <StyledContainer testID="styled-container">
      <StatusBar style="light" />
      <InnerContainer testID="inner-container">
        <WelcomeImage
          resizeMode="cover"
          source={require("./../../assets/ZenTimer6.png")}
          testID="welcome-image"
        />

        <WelcomeContainer testID="welcome-container">
          <PageTitle welcome={true} testID="welcome-title">
            Welcome!
          </PageTitle>
          <SubTitle welcome={true} testID="user-name">
            Zen User
          </SubTitle>
          <SubTitle welcome={true} testID="user-email">
            serenity@gmail.com
          </SubTitle>
          <StyledFormArea>
            <Avatar
              resizeMode="cover"
              source={require("./../../assets/logoZenTimer.png")}
              testID="avatar-image"
            />
            <Line testID="line" />
            <StyledButton testID="logout-styled-button">
              <ButtonText testID="logout-button-text">Logout</ButtonText>
            </StyledButton>
          </StyledFormArea>
        </WelcomeContainer>
      </InnerContainer>
    </StyledContainer>
  );
};

export default WelcomeScreen;
