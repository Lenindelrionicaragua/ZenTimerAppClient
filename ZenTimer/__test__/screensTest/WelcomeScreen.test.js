import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, act, waitFor } from "@testing-library/react-native";

import WelcomeScreen from "../../screens/WelcomeScreen/WelcomeScreen";

import { StatusBar } from "react-native";
import {
  InnerContainer,
  PageLogo,
  PageTitle,
  SubTitle,
  WelcomeContainer,
} from "../../screens/WelcomeScreen/WelcomeScreenStyles";

// Rendering Functions
const renderWelcomeScreen = () => render(<WelcomeScreen />);
const renderWelcomeScreenWithRenderer = () =>
  renderer.create(<WelcomeScreen />);

let welcomeScreenRender;
let welcomeScreenRenderWithRenderer;

beforeEach(() => {
  welcomeScreenRender = renderWelcomeScreen();
  welcomeScreenRenderWithRenderer = renderWelcomeScreenWithRenderer();
});

//SignupScreen
describe("WelcomeScreen", () => {
  test("Renders correctly the WelcomeScreen Component", () => {
    const WelcomeScreenSnapshot = welcomeScreenRenderWithRenderer.toJSON();
    expect(WelcomeScreenSnapshot).toMatchSnapshot();
  });

  test("Render StatusBar correctly", () => {
    const WelcomeScreenInstance = welcomeScreenRenderWithRenderer.root;
    const statusBar = WelcomeScreenInstance.findByType(StatusBar);
    expect(statusBar).toBeTruthy();
  });

  test("Render UI components correctly", () => {
    const { getByTestId } = welcomeScreenRender;
    expect(getByTestId("welcome-container")).toBeTruthy();
    expect(getByTestId("inner-container")).toBeTruthy();
    expect(getByTestId("welcome-title")).toBeTruthy();
    expect(getByTestId("user-name")).toBeTruthy();
  });

  test("PageTitle should render a string of letters, numbers, or spaces", () => {
    const { getByTestId } = welcomeScreenRender;
    const pageTitleComponent = getByTestId("welcome-title");
    const textContent = pageTitleComponent.props.children.toString();
    expect(textContent).toMatch("Welcome! Prix");
  });

  test("SubTitle should render a string of letters, numbers or spaces", () => {
    const { getByTestId } = welcomeScreenRender;
    const subTitleComponent = getByTestId("user-name");
    const textContent = subTitleComponent.props.children.toString();
    expect(textContent).toMatch("Zen User");
  });
});

// WelcomeContainer Integration Tests
describe("WelcomeContainer Integration Tests", () => {
  let welcomeComponent;

  beforeEach(() => {
    const WelcomeScreenInstance = welcomeScreenRenderWithRenderer.root;
    welcomeComponent = WelcomeScreenInstance.findByType(WelcomeContainer);
  });

  test("Render a WelcomeContainer component", () => {
    expect(welcomeComponent).toBeTruthy();
  });

  // Login StyledButton
  describe("StyledButton", () => {
    test("Render StyledButton", () => {
      const { getByTestId } = welcomeScreenRender;
      const styledButtonElement = getByTestId("login-styled-button");
      expect(styledButtonElement).toBeTruthy();
    });
  });
});
