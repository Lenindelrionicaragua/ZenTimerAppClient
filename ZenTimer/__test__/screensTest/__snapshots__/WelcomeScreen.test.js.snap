// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen Renders correctly the WelcomeScreen Component 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffc500",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingBottom": 21,
      "paddingLeft": 21,
      "paddingRight": 21,
      "paddingTop": 64,
    }
  }
  testID="styled-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffe1a8",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
    testID="inner-container"
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        {
          "height": "50%",
          "width": "100%",
        }
      }
      testID="welcome-image"
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffe1a8",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 25,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 10,
          "width": "100%",
        }
      }
      testID="welcome-container"
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 35,
            "fontWeight": "bold",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "textAlign": "center",
          }
        }
        testID="welcome-title"
        welcome={true}
      >
        Welcome! Prix
      </Text>
      <Text
        style={
          {
            "color": "#1a1204",
            "fontSize": 10,
            "fontWeight": "normal",
            "letterSpacing": 1,
            "marginBottom": 5,
          }
        }
        testID="user-name"
        welcome={true}
      >
        Zen User
      </Text>
      <Text
        style={
          {
            "color": "#1a1204",
            "fontSize": 10,
            "fontWeight": "normal",
            "letterSpacing": 1,
            "marginBottom": 5,
          }
        }
        testID="user-email"
        welcome={true}
      >
        serenity@gmail.com
      </Text>
      <View
        style={
          {
            "width": "90%",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            {
              "borderBottomLeftRadius": 50,
              "borderBottomRightRadius": 50,
              "borderColor": "#ffc500",
              "borderTopLeftRadius": 50,
              "borderTopRightRadius": 50,
              "borderWidth": 2,
              "height": 100,
              "marginBottom": 10,
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 10,
              "width": 100,
            }
          }
          testID="avatar"
        />
        <View
          style={
            {
              "backgroundColor": "#ffc500",
              "height": 1,
              "marginVertical": 10,
              "width": "100%",
            }
          }
          testID="line"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "height": 60,
              "justifyContent": "center",
              "marginVertical": 5,
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
          testID="login-styled-button"
        >
          <Text
            style={
              {
                "color": "#ffe1a8",
                "fontSize": 16,
              }
            }
            testID="login-button-text"
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
